conda/__init__.py

```diff
@@ -148,3 +148,85 @@ def _default(self, obj):
 
 _default.default = JSONEncoder().default
 JSONEncoder.default = _default
+
+
+# inspired by deprecation (https://deprecation.readthedocs.io/en/latest/) and
+# CPython's warnings._deprecated
+class _deprecated:
+    def __init__(
+        self,
+        deprecate_in: str,
+        remove_in: str,
+        addendum: str | None = None,
+        _current: str | None = None,
+    ):
+        """
+        Args:
+            deprecate_in: Version in which code will be marked as deprecated.
+            remove_in: Version in which code is expected to be removed.
+            addendum: Additional messaging. Useful to indicate what should be used instead.
+            _current: Testing variable, shouldn't be used in practice.
+        """
+        from packaging import version
```

Comment:
🤷🏻‍♂️ because conda's version should always be PEP440? But nothing wrong with using `conda.models.version.normalized_version` instead especially given the speed difference:

```
python -m timeit -s 'from conda.models.version import normalized_version as parse' -- 'parse("22.3.0+93_kep98nrt_py0")'
1000000 loops, best of 5: 293 nsec per loop

python -m timeit -s 'from packaging.version import parse' -- 'parse("22.3.0+93_kep98nrt_py0")'
1000000 loops, best of 5: 8.84 usec per loop
```

Unfortunately, importing from `conda.models.version` results in a circular import, so perhaps this isn't worthwhile.

Reply:
Ended up implementing a dumb parser: https://github.com/conda/conda/pull/12125/commits/61dc68a73efc3e138038a81858c83eb2c0d82086